packages/forma-36-codemod/bin/updateDependencies.js

```diff
@@ -0,0 +1,133 @@
+const path = require('path');
+const fs = require('fs');
+const chalk = require('chalk');
+const execa = require('execa');
+const semverSatisfies = require('semver/functions/satisfies');
+const readPkgUp = require('read-pkg-up');
+const packages = require('./packages-list');
+const inquirer = require('inquirer');
+
+const dependencyProperties = [
+  'dependencies',
+  'devDependencies',
+  'clientDependencies',
+  'isomorphicDependencies',
+  'buildDependencies',
+];
+
+const getCmd = (pkgManager, cmd, pkgs) => {
+  const cmds = {
+    npm: {
+      install: `npm install -S ${pkgs.join(' ')}`,
+      uninstall: `npm uninstall ${pkgs.join(' ')}`,
+    },
+    yarn: {
+      install: `yarn add ${pkgs.join(' ')}`,
+      uninstall: `yarn remove ${pkgs.join(' ')}`,
+    },
+  };
+  return pkgs.length > 0 ? cmds[pkgManager][cmd] : '';
+};
+
+function getOutput(pkgManager, { newPackages, removePackages }) {
+  if (pkgManager.length === 0) process.exit(0);
+  console.log('You can run the following commands to update the packages:\n');
+  pkgManager.forEach((pkgManager) => {
+    const result = [
+      chalk.bold(`For ${pkgManager}:`),
+      `  ${getCmd(pkgManager, 'uninstall', removePackages)}`,
+      `  ${getCmd(pkgManager, 'install', newPackages)}`,
+    ];
+    console.log(result.filter((r) => r.trim()).join('\n'), '\n');
+  });
+}
+
+async function updateDependencies(targetDir) {
+  const cwd = path.resolve(process.cwd(), targetDir);
+  const closestPkgJson = readPkgUp.sync({ cwd });
+  let pkgManager = ['npm', 'yarn'];
+  let removePackages = packages.PACKAGES_REMOVE;
+  let newPackages = Object.keys(packages.PACKAGES_UPGRADE);
+
+  if (!closestPkgJson) {
+    console.log(
+      chalk.yellow(
+        "We didn't find your package.json, you may need to remove or install some packages:",
+      ),
+    );
+    newPackages = newPackages.map(
+      (pkg) => `${pkg}@${packages.PACKAGES_UPGRADE[pkg]}`,
+    );
+
+    return getOutput(pkgManager, { newPackages, removePackages });
+  }
+  const pkgDirname = path.dirname(closestPkgJson.path);
+  const { packageJson } = closestPkgJson;
+
+  // Check if packages need to be upgraded or removed
+  const allDeps = dependencyProperties.reduce(
+    (acc, dependency) => Object.assign(acc, packageJson[dependency] || {}),
+    {},
+  );
+  removePackages = removePackages.filter((pkg) => Boolean(allDeps[pkg]));
+  newPackages = newPackages
+    .filter(
+      (pkg) =>
+        !allDeps[pkg] ||
+        !semverSatisfies(packages.PACKAGES_UPGRADE[pkg], allDeps[pkg]),
+    )
+    .map((pkg) => `${pkg}@${packages.PACKAGES_UPGRADE[pkg]}`);
+
+  // Select package manager based on lock file, fallback to npm if no lock file is found
+  pkgManager = [
+    (fs.existsSync(path.resolve(pkgDirname, 'package-lock.json')) && 'npm') ||
+      (fs.existsSync(path.resolve(pkgDirname, 'yarn.lock')) && 'yarn') ||
+      'npm',
+  ];
```

Comment:
Didn't get why u used an array here? It will have only one item always, isn't it?

And another comment is that I would move `(fs.existsSync(path.resolve(pkgDirname, 'yarn.lock')) && 'yarn')` before check with `npm`, since if there both lock files, we should give a priority to `yarn`. We have such a case in our `field-editors`.

Reply:
The array is mostly so we don't need to check the data type of the pkgManager on the getOutput function, we just use a forEach, and it will print just one or both package manager commands.

We can move, but I would still fallback to `npm` in case no lock file is found, as it's more probable that the user would have npm installed. WDYT??

Reply:
Yes, of course, fallback to `npm` is totally valid.